solidity/src/base/Transcript.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: UNLICENSED
+// This is licensed under the Cryptographic Open Software License 1.0
+pragma solidity ^0.8.28;
+
+import "./Constants.sol"; // solhint-disable-line no-global-import
+
+library Transcript {
```

Comment:
What does this do?

Can we have a README that explains the different libraries in this PR and what they're used for?